staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers_test.go

```diff
@@ -225,3 +225,80 @@ func TestSetNestedMap(t *testing.T) {
 	assert.Len(t, obj["x"].(map[string]interface{})["z"], 1)
 	assert.Equal(t, obj["x"].(map[string]interface{})["z"].(map[string]interface{})["b"], "bar")
 }
+
+func TestNestedValueNoCopy(t *testing.T) {
+	type fooStruct struct {
+		Foo int
+		Bar bool
+	}
+
+	obj := map[string]interface{}{
+		"a": map[string]interface{}{
+			"b": 1,
+			"c": "c",
+			"d": []int{1, 2, 3},
+			"e": map[string]string{"foo": "bar"},
```

Comment:
nit: while this works since you are only traversing as far as `e`, but I think this should be `map[string]any` since `map[string]string` is not explicitly supported as unstructured. Same for `[]int`.